Vaults/Nodsoft.MoltenObsidian.Vaults.Http/VaultDependencyInjectionExtensions.cs

```diff
@@ -23,14 +24,8 @@ public static class VaultDependencyInjectionExtensions
 	public static IServiceCollection AddMoltenObsidianHttpVault(this IServiceCollection serviceCollection, Func<IServiceProvider, HttpClient> httpClientProvider)
 		=> serviceCollection.AddSingleton<IVault>(services =>
 		{
-			HttpClient httpClient = httpClientProvider(services);
-
-			// Get the vault manifest from the server
-			RemoteVaultManifest manifest = httpClient.GetFromJsonAsync<RemoteVaultManifest>("moltenobsidian.manifest.json").GetAwaiter().GetResult()
-				?? throw new InvalidOperationException("Failed to retrieve the vault manifest from the server.");
-
-			return HttpRemoteVault.FromManifest(manifest, httpClient);
-		});
+			return services.GetRequiredService<Task<IVault>>().Result;
+		}).AddMoltenObsidianHttpAsyncVault(httpClientProvider);
```

Comment:
Going on a whim with this one, are we sure this one will return the same Vault instance each time ? 
(I don't see why it shouldn't, but just making sure)